tensorflow/contrib/tensorrt/convert/convert_graph.cc

```diff
@@ -878,10 +893,11 @@ tensorflow::Status ConvertAfterShapes(ConversionParams& params) {
       continue;
     }
     curr_engine.precision_mode = params.precision_mode;
-    curr_engine.engine_type =
-        (params.is_dyn_op || params.precision_mode == INT8MODE
+    curr_engine.engine_type = ((params.is_dyn_op ||
+      (params.precision_mode == INT8MODE && params.use_calibration))
```

Comment:
Is precision_mode important? It looks to me that, when use_calibration is true, you should create a dynamic engine regardless of precision_mode.

Reply:
We are ignoring the use_calibration parameter when precision is not INT8 because calibration doesn't make sense for FP32 or FP16. But if you are talking about in the future when we have INT4 then I can see the issue.

Reply:
Then should we explicitly check against calibration in fp? E.g. 

```
if (use_calibration and not int8) {
  LOG(FATAL) << "not implemented";
}
engine_type = (is_dyn_op || use_calibration) ? dynamic : static;
```

Reply:
Good idea, we should do that. In that case, should use_calibration be false by default? That would cause the default INT8 behavior to change, is that okay?

Reply:
I'm totally fine with calibration off by default. In fact, Waymo doesn't use calibration for now.

Reply:
Hmm I think that will break existing users who uses the INT8 api. Maybe just ignore `use_calibration` when precision_mode is not INT8, and mentioned the behavior in the comment/doc-string of `create_inference_graph()`.

Reply:
As a general rule of flags, you shouldn't set a flag true by default if it doesn't work with even the default configuration (in this case `precision=fp32). So I believe use_calibration=false by default is the right thing to do. 

IMO, breaking existing users is fine since this is in tf.contrib. Are you aware of any users on int8 calibration? We could notify them about this change.

Reply:
Btw, not too strong opinion on this. This is a typical trade-off between API stability and future maintenance. @aaroey should make the final call because you are going to maintain this from TF side. 

I'm fine with
```
// This check should appear very early in ConvertGraphDefToTensorRT. 
if (use_calibration and not int8) {
  LOG(ERROR) << "not implemented. fall back to no calibration";
  use_calibration = false;
}

engine_type = (is_dyn_op || use_calibration) ? dynamic : static;
```